architect_py/async_client.py

```diff
@@ -401,6 +401,61 @@ async def send_limit_order(
         )
 
         return await self.get_order(order)
+    
+    async def send_market_pro_order(
+        self,
+        *,
+        market: str,
+        dir: OrderDirection,
+        quantity: DecimalLike,
+        time_in_force_instruction: CreateTimeInForceInstruction = CreateTimeInForceInstruction.DAY,
+        account: Optional[str] = None,
+        source: OrderSource = OrderSource.API,
+        percent_through_market: Decimal = 0.02
+    ) -> Optional[GetOrderOrder]:
+
+        # Check for GQL failures
+        bbo_snapshot = await self.get_market_snapshot(market)
+        if bbo_snapshot is None:
+            raise ValueError("Failed to send market order with reason: no market snapshot for {market}")
+        
+        market_details = await self.get_market(market)
+        if market_details is None:
+            raise ValueError("Failed to send market order with reason: no market details for {market}")
+
+        best_bid = float(bbo_snapshot.bid_price)
+        best_ask = float(bbo_snapshot.ask_price)
```

Comment:
@andrew-lee-architect 
hate to do this after PR is accepted already, but this isn't guaranteed to work